src/CMakeLists.txt

```diff
@@ -27,6 +27,9 @@ if(CMAKE_CXX_COMPILER_ID MATCHES GNU)
     if(ESPP_WALL)
         target_compile_options(_espressopp PRIVATE -Wall)
     endif()
+    if(ESPP_WERROR AND CMAKE_CXX_COMPILER_VERSION VERSION_EQUAL 12.1.1 AND CMAKE_BUILD_TYPE STREQUAL "Release")
+        target_compile_options(_espressopp PRIVATE "-Wno-nonnull")
```

Comment:
I am not sure if I like this. Perhaps it is really an error.

Reply:
I don't like it either but with it being a GCC bug we might need to use this for now.